Web.md

```diff
@@ -123,6 +123,69 @@ MIME type mismatch or `opaque` response types
+match, the locations are easily understandable and the same across browsers.
+
+To achive the same goal of a common representations for WebAssembly constructs, the
+following conventions are adopted.
+
+A WebAssembly location is a reference to a particular instruction in the binary, and may be
+displayed by a browser or engine in similar contexts as JavaScript source locations.
+It has the following format:
+
+`${url}:wasm-function[${funcIndex}]:${pcOffset}`
+
+Where
+* `${url}` is the URL associated with the module, if applicable (see notes).
+* `${funcIndex}` is an index in the [function index space](Modules.md#function-index-space).
+* `${pcOffset}` is the offset in the module binary of the first byte
+  of the instruction, printed in hexadecimal with lower-case digits,
+  with a leading `0x` prefix.
+
+Notes:
+* The URL field may be interpreted differently depending on the
+context. When the response-based
+instantiation [API](#additional-web-embedding-api) is used in a
+browser, the associated URL should be used; or when the
+ArrayBuffer-based instantiation
+[API](JS.md#webassemblyinstantiate) is used, the browser should represent
+the location of the API call. This kind of instantiation is analagous to
+executing JavaScript using `eval`; therefore if the browser has an existing
+method to represent the location of the `eval` call it can use a similar
+one for `WebAssembly.instantiate`. For example if the browser uses
+`foo.js line 10 > eval` or `eval at bar (foo.js:10:3)` for `eval`, it could
+use `foo.js line 10 > WebAssembly.instantiate` or
+`WebAssembly.instantiate at bar (foo.js:10:3)`, respectively.
+Offline tools may use a filename instead.
+* Using hexadecimal for module offsets matches common conventions in native tools
+such as objdump (where addresses are printed in hex) and makes them visually
+distinct from JavaScript line numbers. Other numbers are represented in decimal.
+
+Names of functions may also be displayed if the module contains a
+["name" section](BinaryEncoding.md#name-section);
+these can be used in the same contexts as JavaScript functions.
+If the module's name section includes both module and function names, it
+should be represented as `${module_name}.${function_name}`.
+If the function name is absent, then the output can be context-dependent.
+For example, if the function name is shown alongside its location in a
+stack trace, then just the module name (if present) or an empty string
+can be used, because the function index is shown with the location.
+Otherwise a representation that indicates the function index should be
+used (for example `${module_name}.wasm-function[${funcIndex}]`).
+
+Note also that this document does
+not specify the full format of strings such as stack frame representations;
+this allows engines to continue using their existing formats for JavaScript
+(which existing code may already be depending on) while still printing
+WebAssembly frames in a format consistent with JavaScript.
```

Comment:
Yeah, that makes sense if client code would otherwise be doing their own binary parsing that we've already done.  With the other Module reflection methods, our motivating use case was module loaders (and specific experience with incorporating wasm into SystemJS).  It'd be nice to have some specific user who is wanting to programmatically access these function names.

Anyhow, this probably belongs in a different issue.